From: Xiuling Ma <xma@us.ibm.com>
Date: Mon, 13 Jan 2014 11:57:03 -0500
Subject: disable catas_reset by default to avoid problems with EEH
Patch-mainline: Not yet, see below
References: bnc#456389

PPC machines with EEH and Mellanox ib/net cards with catastrophic error
recovery that encounter a PCI bus error can crash and become
unresponsive.

Disable the card reset to avoid this.

NOTE: an upstream fix will come later once IBM can review a couple of
approaches I suggested since this fix is brute force. This driver didn't have
this reset on error feature in SLES10 so it isn't a feature removal.

Signed-off-by: Xiuling Ma <xma@us.ibm.com>
Acked-by: Brandon Philips <bphilips@suse.de>
Acked-by: Benjamin Poirier <bpoirier@suse.de>
---
 drivers/infiniband/hw/mthca/mthca_catas.c  |    2 +-
 drivers/net/ethernet/mellanox/mlx4/catas.c |    4 ++--
 2 files changed, 3 insertions(+), 3 deletions(-)
--- a/drivers/infiniband/hw/mthca/mthca_catas.c
+++ b/drivers/infiniband/hw/mthca/mthca_catas.c
@@ -52,7 +52,7 @@ static LIST_HEAD(catas_list);
 static struct workqueue_struct *catas_wq;
 static struct work_struct catas_work;
 
-static int catas_reset_disable;
+static int catas_reset_disable = 1;
 module_param_named(catas_reset_disable, catas_reset_disable, int, 0644);
 MODULE_PARM_DESC(catas_reset_disable, "disable reset on catastrophic event if nonzero");
 
--- a/drivers/net/ethernet/mellanox/mlx4/catas.c
+++ b/drivers/net/ethernet/mellanox/mlx4/catas.c
@@ -45,11 +45,11 @@ static DEFINE_SPINLOCK(catas_lock);
 static LIST_HEAD(catas_list);
 static struct work_struct catas_work;
 
-static int internal_err_reset = 1;
+static int internal_err_reset = 0;
 module_param(internal_err_reset, int, 0644);
 MODULE_PARM_DESC(internal_err_reset,
 		 "Reset device on internal errors if non-zero"
-		 " (default 1, in SRIOV mode default is 0)");
+		 " (default 0, in SRIOV mode default is 0)");
 
 static void dump_err_buf(struct mlx4_dev *dev)
 {
